tutorials/07-fdem/plot_inv_1_em1dfm.py

```diff
@@ -0,0 +1,350 @@
+survey = fdem.survey.Survey(source_list)
+
+
+###############################################################
+# Assign Uncertainties and Define the Data Object
+# -----------------------------------------------
+#
+# Here is where we define the data that are inverted. The data are defined by
+# the survey, the observation values and the uncertainties.
+#
+
+# 5% of the absolute value
+uncertainties = 0.05*np.abs(dobs)*np.ones(np.shape(dobs))
+
+# Define the data object
+data_object = data.Data(survey, dobs=dobs, noise_floor=uncertainties)
+
+
+###############################################################
+# Defining a 1D Layered Earth (1D Tensor Mesh)
+# --------------------------------------------
+#
+# Here, we define the layer thicknesses for our 1D simulation. To do this, we use
+# the TensorMesh class.
+#
+
+# Layer thicknesses
+inv_thicknesses = np.logspace(0,1.5,25)
+
+# Define a mesh for plotting and regularization.
+mesh = TensorMesh([(np.r_[inv_thicknesses, inv_thicknesses[-1]])], '0')
+
+
+#################################################################
+# Define a Starting and/or Reference Model and the Mapping
+# --------------------------------------------------------
+#
+# Here, we create starting and/or reference models for the inversion as
+# well as the mapping from the model space to the active cells. Starting and
+# reference models can be a constant background value or contain a-priori
+# structures. Here, the starting model is log(0.1) S/m.
+#
+# Define log-conductivity values for each layer since our model is the
+# log-conductivity. Don't make the values 0!
+# Otherwise the gradient for the 1st iteration is zero and the inversion will
+# not converge.
+
+# Define model. A resistivity (Ohm meters) or conductivity (S/m) for each layer.
+starting_model = np.log(0.1*np.ones(mesh.nC))
+
+# Define mapping from model to active cells.
+model_mapping = maps.ExpMap()
+
+
+#######################################################################
+# Define the Physics using a Simulation Object
+# --------------------------------------------
+#
+
+simulation = fdem.Simulation1DLayered(
+    survey=survey, thicknesses=inv_thicknesses, sigmaMap=model_mapping
+)
+
+
+#######################################################################
+# Define Inverse Problem
+# ----------------------
+#
+# The inverse problem is defined by 3 things:
+#
+#     1) Data Misfit: a measure of how well our recovered model explains the field data
+#     2) Regularization: constraints placed on the recovered model and a priori information
+#     3) Optimization: the numerical approach used to solve the inverse problem
+#
+#
+
+# Define the data misfit. Here the data misfit is the L2 norm of the weighted
+# residual between the observed data and the data predicted for a given model.
+# The weighting is defined by the reciprocal of the uncertainties.
+dmis = data_misfit.L2DataMisfit(simulation=simulation, data=data_object)
+dmis.W = 1./uncertainties
```

Comment:
this line isn't necessary as long as the uncertainties are provided to the data object (which they are in this example). In general, I think we want to be encouraging folks to set this through the data misfit object rather than manually, so I would suggest removing this line

Reply:
good point. removed the line.